### PR TITLE
Autounlock techwebs are shared between machines that use them.

### PR DESCRIPTION
**Pull request by JohnFulpWillard**
## About The Pull Request

Did you know that for every Autolathe, Limbgrower, Biogenerator, ORM, and Smelter that was built, an entire new techweb was made? The average round has 2 ORMs (smelters count) if not more from ORM deconstruction objective, 2 biogenerators, several autolathes and generally 0-1 limbgrower, I think there's more techwebs being created than we need.

Creating a whole techweb was a pretty terrible way to optimize this, so I made a global list that stores these techwebs. Created on demand, these autounlocking techwebs now share between the machines that use them. It also generate all hacked nodes which will be visible if the machine is hacked, instead of 'researching' them for each hacked individual machine.

The 'specialized' techweb subtype was removed because its sole purpose was to allow autounlocking to be a subtype of it. Now autounlocking is just the direct subtype. I also removed mechfab autounlocking type because it wasn't used, mechfabs use techwebs directly.

Autolathes and Limbgrowers now locally store 'imported designs' which are things uploaded from technology disks. Outside of this, the autounlocking techweb subtype now stores 'hacked designs' which unlocks when the machine is emagged.
While doing this, I saw ORMs and Biogenerators had disks you can insert into them, but I did not find anything that can actually be uploaded to them (I saw Alien Alloys as a possibility, but there's no such disk to allow uploading this alloy to the machine, and I didn't think an entire system for 1 single alloy was worth keeping around), so I removed this unused feature from both machines.

I merged 'Hacked' and 'Emagged' categories because they served the same purpose, 'hacked' being on hacked autolathes and 'emagged' being on hacked (emagged) limbgrowers.

Tech disk techwebs (which is created every time a disk is made, however I hope to change this in the future) into it's own subtype, so admins understand where its from when looking in VV panel (because I was confused when I saw them at first).

``autolathe_crafted`` proc was removed because it was entirely unused too.

Now it looks like this, which I consider an improvement:
image

## Why It's Good For The Game

We no longer initialize a brand new techweb for no reason whatsoever.
Each techweb made is making entire lists of experiments and research papers, all of which is never to be seen in-game and is completely useless to the player. Cutting down on these techwebs is a good first step to this needless initializing.

## Changelog

:cl:
qol: Removed the tech disk part of the ORM's UI as it was entirely unused, now it's a little more compact with less scrolling needed.
refactor: Autolathes, Limb growers, Biogenerators, ORMs and Smelters now share techwebs with other machines of their types, rather than all make new techwebs each time. This means they only build their nodes once, including hacked ones. Instead of researching nodes on hacking the machine, they now show hacked ones depending on if it's hacked. 
/:cl:

### CODE REVIEW

File: code/modules/research/techweb/all_nodes.dm [hunk 1]
```diff
@@ -2285,26 +2285,3 @@
...
-	for(var/i in subtypesof(/datum/techweb_node))
-		processing += new i
-	var/datum/techweb/TW = new
-	TW.research_points = list()
-	for(var/i in processing)
-		var/datum/techweb_node/TN = i
-		TW.add_point_list(TN.research_costs)
-	return TW.research_points
-
-/proc/total_techweb_points_printout()
```
Comment (JohnFulpWillard) [inline code/modules/research/techweb/all_nodes.dm:2301]: I was checking where techwebs are created and found these two procs which did nothing, so I just decided to remove them.

**Comment by ZephyrTFA** [comment]
And how will this interact with machines being EMAGged. Won't this lead to emagging one emagging all similar types

**Comment by JohnFulpWillard** [comment]
> And how will this interact with machines being EMAGged. Won't this lead to emagging one emagging all similar types

True, thanks for catching that

File: code/game/machinery/limbgrower.dm [hunk 2]
```diff
@@ -63,7 +75,14 @@
 	var/species_categories = categories.Copy()
 	for(var/species in species_categories)
 		species_categories[species] = list()
-	for(var/design_id in stored_research.researched_designs)
+
+	var/list/available_nodes = stored_research.researched_designs
```
Comment (ZephyrTFA) [inline code/game/machinery/limbgrower.dm:79]:
```suggestion
	var/list/available_nodes = stored_research.researched_designs.Copy()
```

**Review by ZephyrTFA — CHANGES_REQUESTED** [review]

**Review by ZephyrTFA — APPROVED** [review]
Looks good to me, currently cannot test it locally but I see no glaring issues

File: code/modules/research/techweb/techweb_types.dm [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+ * Admin techweb that has everything unlocked by default
+ */
+/datum/techweb/admin
+	id = "ADMIN"
+	organization = "CentCom"
+
+/datum/techweb/admin/New()
+	. = ..()
+	for(var/i in SSresearch.techweb_nodes)
+		var/datum/techweb_node/TN = SSresearch.techweb_nodes[i]
```
Comment (jlsnow301) [inline code/modules/research/techweb/techweb_types.dm:24]: can you use `as anything` here

**Review by jlsnow301 — CHANGES_REQUESTED** [review]

File: code/modules/research/techweb/techweb_types.dm (hunk 3 above)
Comment (JohnFulpWillard) [reply code/modules/research/techweb/techweb_types.dm:24]: We don't need ``as anything`` because we're not casting any type as ``i``

Comment (jlsnow301) [reply code/modules/research/techweb/techweb_types.dm:24]: sry i mean a for .. in loop rather, `for(var/datum/techweb_node/node in SSresearch.techweb_nodes)`

Comment (ZephyrTFA) [reply code/modules/research/techweb/techweb_types.dm:24]: No we can't, because it's an assosciate list

**Pull request merged**